libs/utils/include/utils/Zip2Iterator.h

```diff
@@ -60,11 +60,9 @@ class Zip2Iterator {
     Zip2Iterator(Zip2Iterator const& rhs) noexcept = default;
     Zip2Iterator& operator=(Zip2Iterator const& rhs) = default;
 
-    value_type operator*() const { return { *mIt.first, *mIt.second }; }
-    reference  operator*()       { return { *mIt.first, *mIt.second }; }
+    reference operator*() const { return { *mIt.first, *mIt.second }; }
```

Comment:
shouldn't this be a `const_reference`?

Reply:
No, that would only be if this were `ConstZip2Iterator`